src/ansys/dpf/core/results.py

```diff
@@ -520,6 +527,52 @@ def on_mesh_scoping(self, mesh_scoping):
         self._mesh_scoping = mesh_scoping
         return self
 
+    def on_region_scoping(self, region_scoping):
+        """Set the mesh scoping to a given mesh scoping.
+
+        Parameters
+        ----------
+        region_scoping : Scoping, list[int]
+            Scoping identifying a list of regions (part IDs, zone IDs...)
+
+        Returns
+        -------
+        self : Result
+
+        Examples
+        --------
+        Use a list of fluid face zones.
+
+        >>> from ansys.dpf import core as dpf
+        >>> from ansys.dpf.core import examples
+        >>> model = dpf.Model(examples.download_fluent_mixing_elbow_steady_state())
+        >>> disp = model.results.displacement
+        >>> fc = disp.on_region_scoping([1,2,3]).eval()
+        >>> len(fc[0].scoping)
+        3
+
+        Use a scoping to specify a list of entity IDs with their locations.
+
+        >>> stress = model.results.stress
+        >>> scop = dpf.Scoping(ids=[3,4,5], location= dpf.locations.zone)
+        >>> fc = stress.on_mesh_scoping(scop).eval()
+        >>> len(fc[0].scoping)
+        3
+        >>> fc[0].location
+        'Faces'
+
+        """
+        from ansys.dpf.core import locations
+        if isinstance(region_scoping, list):
+            region_scoping = Scoping(
+                ids=region_scoping,
+                location=locations.zone,
```

Comment:
If the location "zone" is the default, this is only going to be useful for fluids.